e2e-tests/ldap-tls/run

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+
+set -o errexit
+
+test_dir=$(realpath "$(dirname "$0")")
+. "${test_dir}/../functions"
+set_debug
+
+deploy_openldap() {
+	yq "$test_dir/conf/openldap.yaml" \
+		| yq "select(.metadata.name == \"ldap-ca\").spec.dnsNames[0]=\"openldap.$namespace.svc.cluster.local\"" \
+		| kubectl_bin apply -f -
+
+	kubectl rollout status deployment/openldap --timeout=120s
+}
+
+test_mongod_openldap() {
+	cluster="some-name"
+
+	kubectl_bin create secret generic $cluster-rs0-mongod --from-literal=mongod.conf="$(yq "$test_dir/conf/mongod.conf" \
+		| yq ".security.ldap.servers=\"openldap.$namespace.svc.cluster.local:1636\"")"
+
+	desc "create first PSMDB cluster $cluster"
+	apply_cluster "$test_dir/conf/$cluster.yml"
+
+	desc 'check if all 3 Pods started'
+	wait_for_running $cluster-rs0 3
+
+	run_mongo \
+		'db.getSiblingDB("admin").createRole(
+{
+ role: "cn=admin,ou=perconadba,dc=ldap,dc=local",
+ privileges: [],
+ roles : [
+   {
+     "role" : "readAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "dbAdminAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "clusterMonitor",
+     "db" : "admin"
+   },
+   {
+     "role" : "readWriteAnyDatabase",
+     "db" : "admin"
+   },
+   {
+     "role" : "restore",
+     "db" : "admin"
+   },
+   {
+     "role" : "backup",
+     "db" : "admin"
+   }
+ ]})' \
+		"userAdmin:userAdmin123456@$cluster-rs0.$namespace"
+
+	run_mongo 'db.runCommand({connectionStatus:1})' "percona:password@$cluster-rs0.$namespace" "" "" "--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'"
+
+	roles_size=$(run_mongo 'JSON.stringify(db.runCommand({connectionStatus:1}))' \
+		"percona:password@$cluster-rs0.$namespace" "" "" \
+		"--authenticationMechanism 'PLAIN' --authenticationDatabase '\$external'" \
+		| grep -E -v "Percona Server for MongoDB|connecting to:|Implicit session:|versions do not match|Error saving history file:|bye" \
+		| jq '.authInfo.authenticatedUserRoles | length')
+
+	if [ -z "$roles_size" ]; then
```

Comment:
it would be better if we checked for some string response and not just number of roles

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1480/commits/6e770f75cc0402167333b367a07563bcfba6b2de